libraries/ms-common/src/stm32f0xx/spi.c

```diff
@@ -0,0 +1,185 @@
+#pragma once
+
+#include "spi.h"
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include "gpio.h"
+#include "status.h"
+#include "stm32f0xx.h"
+
+static SPI_TypeDef *spi_periph_map[] = { SPI1, SPI2 };
+
+static uint8_t s_spi_x = 0;
+
+static uint8_t s_tx_buffer = 0;
+static uint8_t s_rx_buffer = 0;
+static bool s_transmitted = 0;
+static bool s_received = 0;
+
+static GPIOAddress s_cs_pin_map[] = { 0, 0 };
+
+static bool prv_is_periph_valid(const uint8_t spi_x) {
+  return spi_x == 0 || spi_x == 1;
+}
+
+static bool prv_are_settings_valid(const SPISettings *settings) {
+  return !(settings->polarity > SPI_CPOL_HIGH ||
+           settings->phase > SPI_CPHA_2EDGE ||
+           settings->baud_rate > SPI_BAUDRATE_256 ||
+           settings->first_bit > SPI_FIRSTBIT_LSB);
+}
+
+static bool prv_is_state_valid(const GPIOState state) {
+  return state < NUM_GPIO_STATE;
+}
+
+static uint8_t prv_exchange(uint8_t data) {
+  s_transmitted = false;
+  s_received = false;
+  s_tx_buffer = data;
+
+  SPI_I2S_ITConfig(spi_periph_map[s_spi_x], SPI_I2S_IT_TXE, ENABLE);
```

Comment:
Just a heads up, you only need to enable the interrupt once. You don't need to disable it every time you're in an interrupt.